lib/containers/common.pm

```diff
@@ -231,7 +231,8 @@ sub test_container_image {
     } else {
         # Pull the image if necessary
         if (script_run("$runtime image inspect --format='{{.RepoTags}}' $image | grep '$image'") != 0) {
-            assert_script_run("$runtime pull $image", timeout => 300);
+            # At least on publiccloud, this image pull can take long and occasinally fails due to network issues
+            script_retry("$runtime pull $image", retry => 3, timeout => 420);
```

Comment:
good for short term , but in long run maybe we need to replace this with something smarter ? like grep'ing actual failure and retry only in case it is what we expect ?
Image pull might fail because of dozen different reasons among them there one which can be ignored with retry but there are also some others which might hide real issues in case we will retry blindly

Reply:
Yes, we can perhaps improve this with a little bit more brain here.